### PR TITLE
With domain

### PR DESCRIPTION
**Pull request by mikkelspring**

### CODE REVIEW

File: src/main/scala/ylabs/messaging/Client.scala [hunk 1]
```diff
@@ -62,6 +62,16 @@ class Client extends FSM[State, Context] {
   lazy val domain = config.getString(ConfigKeys.domain)
   lazy val host = config.getString(ConfigKeys.host)
 
+  def splitUserIntoNameAndDomain(user: User):(String, String) = {
```
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala]: to be consistent with the rest of the codebase you could return `(User, Domain)`, i.e. create a `case class Domain(value: String) extends AnyVal`

File: src/main/scala/ylabs/messaging/Client.scala [hunk 2]
```diff
@@ -62,6 +62,16 @@ class Client extends FSM[State, Context] {
...
+    val (u ,_) = user.value.span(c => c != '@')
+    (u, domain) //for now only use the domain in the configuration regardless of what the user types
```
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala]: i don't understand that comment..?

File: src/main/scala/ylabs/messaging/Client.scala [hunk 3]
```diff
@@ -62,6 +62,16 @@ class Client extends FSM[State, Context] {
...
+    // (u, if(d.isEmpty()) domain else d  )
```
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala]: delete this line?

File: src/test/scala/ylabs/messaging/ClientTest.scala [hunk 4]
```diff
@@ -149,6 +152,130 @@ class ClientTest extends WordSpec with Matchers with BeforeAndAfterEach {
     }
   }
 
+  //Domain added to user name
+  "with user domain connects to the xmpp server"  in new FixtureWithDomain {
```
Comment (mpollmeier) [inline src/test/scala/ylabs/messaging/ClientTest.scala]: try to not use comments too much - the code should be self explanatory, and comments get often ignored during refactorings

File: src/test/scala/ylabs/messaging/ClientTest.scala [hunk 5]
```diff
@@ -149,6 +152,130 @@ class ClientTest extends WordSpec with Matchers with BeforeAndAfterEach {
...
+      user2 ! Connect(username2, user2Pass)
+      user1Listener.fishForMessage(3 seconds, "notification that user2 came online") {
+        case UserBecameAvailable(user) ⇒
+          user.value should startWith(username2.value)
+          true
+      }
+    }
+  }
+
+  "with user domain connects to the xmpp server provides information about who is online and offline (roster)" in new FixtureWithDomain {
```
Comment (mpollmeier) [inline src/test/scala/ylabs/messaging/ClientTest.scala]: if all your tests start with the same words it's a sign that you should have a hierarchy of tests. e.g.

```
  "bla" when {
    "foo" should {
      "blub" can {
        "bar" in new Fixture {
```

http://doc.scalatest.org/2.0/index.html#org.scalatest.WordSpec

**Comment by mikkelspring** [comment]
OK how does this look? I'm investigating how to call the unit tests with different fixtures but not have to repeat the same test code

**Pull request merged**